src/sync/deployment.ts

```diff
@@ -50,7 +57,19 @@ const getTransformedDeployments = async (deployments, gitHubInstallationClient:
 
 export const getDeploymentTask = async (logger: Logger, gitHubInstallationClient: GitHubInstallationClient, jiraHost: string, repository: Repository, cursor?: string | number, perPage?: number, data?: BackfillMessagePayload) => {
 	logger.debug("Syncing Deployments: started");
-	const { edges, deployments } = await fetchDeployments(gitHubInstallationClient, repository, cursor, perPage);
+
+	const shouldUseIncrementalBackfill = await booleanFlag(BooleanFlags.USE_BACKFILL_ALGORITHM_INCREMENTAL, jiraHost);
+	const { edges, deployments } = await fetchDeployments(shouldUseIncrementalBackfill, gitHubInstallationClient, repository, cursor, perPage);
+
+	if (shouldUseIncrementalBackfill) {
+		const fromDate = data?.commitsFromDate ? new Date(data?.commitsFromDate) : undefined;
+		if (isAllEdgesEarlierThanFromDate(edges, fromDate)) {
```

Comment:
Shouldn't we use graphql  `since` filter rather than manual filtering?

Reply:
Unfortunately the `deployment` in graphql doesn't have `since` 😞